auth/realm/base/src/main/java/org/wildfly/security/auth/realm/AggregateSecurityRealm.java

```diff
@@ -88,12 +101,18 @@ public RealmIdentity getRealmIdentity(final Evidence evidence) throws RealmUnava
     public RealmIdentity getRealmIdentity(final Principal principal) throws RealmUnavailableException {
         boolean ok = false;
         final RealmIdentity authenticationIdentity = authenticationRealm.getRealmIdentity(principal);
+
+        Principal authorizationPrincipal = principal;
+        if(principalTransformer != null) {
+            authorizationPrincipal = principalTransformer.apply(authorizationPrincipal);
+            if (authorizationPrincipal == null) return null;
+        }
+
         final RealmIdentity[] authorizationIdentities = new RealmIdentity[authorizationRealms.length];
         try {
             for (int i = 0; i < authorizationIdentities.length; i++) {
                 SecurityRealm authorizationRealm = authorizationRealms[i];
-
-                authorizationIdentities[i] = authorizationRealm == authenticationRealm ? authenticationIdentity : authorizationRealm.getRealmIdentity(principal);
+                authorizationIdentities[i] = authorizationRealm == authenticationRealm && principalTransformer == null ? authenticationIdentity : authorizationRealm.getRealmIdentity(authorizationPrincipal);
```

Comment:
Just a minor comment, for readability, it might be good to add brackets:

```(authorizationRealm == authenticationRealm) && (principalTransformer == null)```

Reply:
Fixed